src/Raven.Server/Documents/DocumentPutAction.cs

```diff
@@ -212,6 +212,7 @@ public DocumentPutAction(DocumentsStorage documentsStorage, DocumentDatabase doc
             };
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private string BuildDocumentKey(DocumentsOperationContext context, string key, Table table, long newEtag, out bool knownNewKey)
```

Comment:
1. It is looking better this way. (one line less). 
2. @redknightlois warned me not to use Tuple in hot paths, at least until .net core 2.0 is out.